### DISCUSSION TITLE
Integrated AIR desktop update

### DISCUSSION THREAD

**Discussion by 42retfa** [airsdk/Adobe-Runtime-Support#1330] [General]
Hi,

Just a question about the AIR desktop runtime : is the Harman team planning to develop some sort of integrated update system for the AIR applications packaged with the captive runtime option ?

Thank you.

Vincent.

**Comment by ajwfrost** [comment]
Do you mean, updating the runtime library itself? or do you mean, a framework to update your SWF file or other supporting files?

One reason for going for the captive runtime option was to protect from unwanted updates of the runtime: Adobe had found that when a shared runtime release went out, it had the potential to cause problems with existing apps that were installed and used it. So we don't want to be automatically updating the captive runtime that's part of an application. If you wanted to update this then it should be carried out as part of a full application update process.

So then if it's about a way for applications to be updated then this depends a bit on how you package and deploy them. There's an interesting discussion at #1294 which has some suggestions. For our (non-AIR) apps we use Wix and this creates update packages that can be configured so that you only install what you need to install.

I'm wondering whether it would help though if we provided an API in AIR that allowed the application files to be overwritten (often this would mean asking the user for elevated permissions so we could write to "Program Files" or "/Applications") so then you could have your application update itself by downloading a package and extracting that over the top of the existing app. You could perhaps already achieve that if you had an application that stored its main logic and assets within the "app-storage" folder and then in the read-only installation folder it just had a bootstrapper that loaded this, and could update this...

So .. a lot of options!

thanks

**Reply by 42retfa** [reply]
Hi Andrew,

Thank you for your response. 

I didn't meant "updating the runtime library itself" but in fact, as you suggest, providing an API that allowed the application files to be overwritten.

We have already integrated an update system to our desktop AIR applications that allows the ".swf" contents to be overwritten transparently when needed but this system doesn't manage the runtime update. 

For example, the new MacOs Monterey force us to update our desktop application (currently packaged with AIR 31) with the latest AIR version in order to integrate the AIR-5060 bug fix. This means that our client users (5000 students) will have to uninstall and reinstall their application with all the possible issues that can occur during this process.

Same problem if we need to integrate a new native extension to the application.
 
Having the ability to update the runtime transparently would be really helpful (and could save a lot of time to our client customer support team)

Thank you.

Vincent.
